tests/testthat/test-write_output.R

```diff
@@ -0,0 +1,309 @@
+    summarized_file <- file.path(
+      job_id,
+      "summaries",
+      paste0(task_id, ".parquet")
+    )
+    expect_true(file.exists(summarized_file))
+
+    # Check if model rds file was written
+    model_file <- file.path(
+      job_id,
+      "tasks",
+      task_id,
+      "model.rds"
+    )
+    expect_true(file.exists(model_file))
+
+    # Check if metadata JSON file was written
+    metadata_file <- file.path(
+      job_id,
+      "tasks",
+      task_id,
+      "metadata.json"
+    )
+    expect_true(file.exists(metadata_file))
+
+    # Check file contents are right
+    con <- DBI::dbConnect(duckdb::duckdb())
+    on.exit(expr = DBI::dbDisconnect(con))
+    raw_samples_data <- DBI::dbGetQuery(con,
+      "SELECT * FROM read_parquet(?)",
+      params = list(samples_file)
+    )
+    expect_equal(raw_samples_data, mock_samples)
+
+    raw_summaries_data <- DBI::dbGetQuery(con,
+      "SELECT * FROM read_parquet(?)",
+      params = list(summarized_file)
+    )
+    expect_equal(raw_summaries_data, mock_summaries)
+
+
+    written_metadata <- jsonlite::read_json(metadata_file)
+    jsonlite::write_json(mock_metadata, "expected.json")
+    expected_metadata <- jsonlite::read_json(metadata_file)
+
+    expect_equal(written_metadata, expected_metadata)
+  })
+})
+
+test_that("write_model_outputs handles errors correctly", {
+  # Setup: Use an invalid directory to trigger an error
+  invalid_output_dir <- "/invalid/path"
+
+  # Create mock inputs
+  mock_fit <- list(samples = 1:5)
+  mock_metadata <- list(author = "Test", date = Sys.Date())
+
+  # Expect the function to raise a warning due to the invalid directory
+  withr::with_tempdir({
+    expect_warning(
+      write_model_outputs(
+        mock_fit,
+        invalid_output_dir,
+        "job_123",
+        "task_456",
+        mock_metadata
+      ),
+      class = "no_outputs"
+    )
+  })
+})
+
+test_that("write_output_dir_structure generates dirs", {
+  withr::with_tempdir({
+    write_output_dir_structure(".", job_id = "job", task_id = "task")
+
+    expect_true(dir.exists(file.path("job", "tasks", "task")))
+    expect_true(dir.exists(file.path("job", "samples")))
+    expect_true(dir.exists(file.path("job", "summaries")))
+  })
+})
+
+test_that("process_quantiles works as expected", {
+  # Load the sample fit object
+  fit <- readRDS(test_path("data", "sample_fit.rds"))
+
+  # Run the function on the fit object
+  result <- process_quantiles(fit, "test_geo", "test_model", "test_disease")
+
+  # Test 1: Check if the result is a data.table
+  expect_true(
+    data.table::is.data.table(result),
+    "The result should be a data.table"
+  )
+
+  # Test 2: Check if the necessary columns exist in the result
+  expected_columns <- c(
+    "time",
+    "_variable",
+    "value",
+    "_lower",
+    "_upper",
+    "_width",
+    "_point",
+    "_interval",
+    "reference_date",
+    "geo_value",
+    "model",
+    "disease"
+  )
+  expect_equal(
```

Comment:
not 100% necessary, but maybe test if set equal as opposed to "list equal"?

Reply:
https://github.com/CDCgov/cfa-epinow2-pipeline/pull/41/commits/01763b4da0eba1150823fcfda2a57dcf284a42f5